annotator-scanner/src/main/java/edu/ucr/cs/riple/scanner/AnnotatorScanner.java

```diff
@@ -151,6 +169,46 @@ public Description matchMemberSelect(MemberSelectTree tree, VisitorState state)
     return Description.NO_MATCH;
   }
 
+  @Override
+  public Description matchLambdaExpression(
+      LambdaExpressionTree lambdaExpressionTree, VisitorState visitorState) {
+    Config config = context.getConfig();
+    if (!config.callTrackerIsActive()) {
+      return Description.NO_MATCH;
+    }
+    // for e -> Foo.bar(e), assume that method "baz()" has been overridden. Then the containing
+    // method for this lambda is an impacted region for "baz()".  The call to "Foo.bar" is handled
+    // when scanning the body of the lambda.
+    serializeImpactedRegionForFunctionalInterface(config, lambdaExpressionTree, visitorState);
+    return Description.NO_MATCH;
+  }
+
+  @Override
+  public Description matchMemberReference(
+      MemberReferenceTree memberReferenceTree, VisitorState visitorState) {
+    Config config = context.getConfig();
+    if (!config.callTrackerIsActive()) {
+      return Description.NO_MATCH;
+    }
+    // for Foo::bar, which is shorthand for e -> Foo.bar(e), assume that method "baz()" has been
+    // overridden. We need to serialize the impacted region (leaf of path in visitor state)
+    // for both "baz()" and also the called method "bar()".
+    // serialize the overridden method: "baz()"
+    serializeImpactedRegionForFunctionalInterface(config, memberReferenceTree, visitorState);
+    if (memberReferenceTree instanceof JCTree.JCMemberReference) {
+      Symbol calledMethod = ((JCTree.JCMemberReference) memberReferenceTree).sym;
+      if (calledMethod instanceof Symbol.MethodSymbol) {
+        // serialize the called method: "bar()"
```

Comment:
Sorry, one more question here.  Can you give an example of what goes wrong if we _don't_ serialize the called method `bar()` here?  I thought I understood it, but thinking more I'm not so sure.

Reply:
Sure, please consider the code below:
For interface:
```java
interface Foo {
      void m(Object o);
}
```
and code:
```java
void f(Foo f){ }

void baz() {
     f(this::bar);
}
void bar(Object o){ return null; }
```

If we annotate parameter `o` in `Foo#m` as `@Nullable` we will get an error in ` f(this::bar);` since `o` in `m` is `@Nullable` and we are passing it to `bar`. Therefore the region for the member reference expression is a potentially impacted region. Please let me know if this makes sense or I should work on another approach to address this. Thank you.

Reply:
The scenario you describe is related to overriding I think, and is handled by this call, right?

https://github.com/ucr-riple/NullAwayAnnotator/blob/441ec21c492ca72fa8d453518ac2fc8477dc4f24/annotator-scanner/src/main/java/edu/ucr/cs/riple/scanner/AnnotatorScanner.java#L197

You are describing why the method reference needs to be an impacted region for `Foo#m`, which I completely agree with.  I was hoping for an example showing why it _also_ should be an impacted region for `bar`.

Reply:
`serializeImpactedRegionForFunctionalInterface` will only serialize the overriden method (`m` for region `baz()` in this case). But this region is also an impacted region for changes on `bar` as well. Please find the example below:
```java
@Nullable
Object bar(){
    return null;
}

void f(Foo f){}

void baz(){
    f(this::bar);
}
```

This will also trigger an error in `baz()` region since `bar()` is returning `@Nullable`, that without serializing the called method in method reference (`bar()` in this case ) annotator would have missed this error. Please let me know if I understood your question correctly, thank you.

Reply:
I get it now, thanks!  Here, adding a `@Nullable` annotation on the return type of `bar` will cause an override error inside `baz`.  So, `baz` must also be an impacted region for `bar`.  Makes sense